File: modules/objc/flutter/ExtSdkChannelManager.m

#import "ExtSdkChannelManager.h"

NSString *_Nonnull const SEND_CHANNEL = @"com.chat.im.ext.dart_to_native";
NSString *_Nonnull const RECV_CHANNEL = @"com.chat.im.ext.native_to_dart";

@interface ExtSdkChannelManager () {
    NSMutableDictionary *_map;
    NSObject<FlutterPluginRegistrar> *_registrar;
}

@end

@implementation ExtSdkChannelManager

+ (nonnull instancetype)getInstance {
    static ExtSdkChannelManager *instance = nil;
    static dispatch_once_t predicate;
    dispatch_once(&predicate, ^{
      instance = [[ExtSdkChannelManager alloc] init];
    });
    return instance;
}

- (instancetype)init {
    self = [super init];
    if (self) {
        _map = [[NSMutableDictionary alloc] init];
    }
    return self;
}

- (void)setRegistrar:(nonnull NSObject<FlutterPluginRegistrar> *)registrar {
    _registrar = registrar;
}

- (BOOL)add:(nonnull NSString *)name {
    FlutterJSONMethodCodec *codec = [FlutterJSONMethodCodec sharedInstance];
    FlutterMethodChannel *channel =
        [FlutterMethodChannel methodChannelWithName:name
                                    binaryMessenger:[_registrar messenger]
                                              codec:codec];
    [_map setObject:channel forKey:name];
    return YES;
}

- (void)del:(nonnull NSString *)name {
    [_map removeObjectForKey:name];
}

- (nullable FlutterMethodChannel *)get:(nonnull NSString *)name {
    return [_map objectForKey:name];
}

@end

File: native_src/java/com/easemob/ext_sdk/flutter/ExtSdkChannelManager.java
package com.easemob.ext_sdk.flutter;

import androidx.annotation.NonNull;

import java.util.HashMap;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.JSONMethodCodec;
import io.flutter.plugin.common.MethodChannel;

/**
 * 通道管理器
 */
public class ExtSdkChannelManager {

    public class ExtSdkChannelName {
        public static final String SEND_CHANNEL = "com.chat.im.ext.dart_to_native";
        public static final String RECV_CHANNEL = "com.chat.im.ext.native_to_dart";
    }

    private static class SingleHolder {
        static ExtSdkChannelManager instance = new ExtSdkChannelManager();
    }

    public static ExtSdkChannelManager getInstance() {
        return ExtSdkChannelManager.SingleHolder.instance;
    }

    public boolean add(@NonNull BinaryMessenger messenger, @NonNull String name) {
        MethodChannel channel = list.get(name);
        if (channel == null) {
            channel = new MethodChannel(messenger, name, JSONMethodCodec.INSTANCE);
            list.put(name, channel);
            return true;
        }
        return false;
    }
    public void del(@NonNull String name) {
        list.remove(name);
    }
    public MethodChannel get(@NonNull String name) {
        return list.get(name);
    }
    private HashMap<String, MethodChannel> list = new HashMap<>();
}


